### PR TITLE
VT-5650:limit_to_queue_call

### PR DESCRIPTION
**Pull request by ajay-plivo**

### CODE REVIEW

File: sharq_server/server.py [hunk 1]
```diff
@@ -114,14 +112,21 @@ def _view_enqueue(self, queue_type, queue_id):
             'queue_id': queue_id
         })
 
-        try:
-            response = self.sq.enqueue(**request_data)
-        except Exception as e:
-            traceback.print_exc()
-            response['message'] = e.message
-            return jsonify(**response), 400
+        max_queued_length = request_data['payload'].get('max_queued_length', 7200)
```
Comment (manish-plivo) [inline sharq_server/server.py:115]: if max_queued_length is not being passed then this feature should be tuned off. Only if this parameter is part of request, the call queue limit should be enforced

File: requirements.txt [hunk 2]
```diff
@@ -9,5 +9,5 @@ gunicorn==19.0.0
 itsdangerous==0.24
 msgpack==0.5.6
 ujson==2.0.0
-uWSGI==2.0.21
-SharQ==1.2.0
+uWSGI==2.0.19.1
```
Comment (manish-plivo) [inline requirements.txt:12]: why uwsgi version is being downgraded?

File: sharq_server/server.py [hunk 3]
```diff
@@ -114,14 +112,35 @@ def _view_enqueue(self, queue_type, queue_id):
...
+                except Exception as e:
+                    traceback.print_exc()
+                    response['message'] = e.message
+                    return jsonify(**response), 400
+
+                return jsonify(**response), 201
+            else:
+                print("Max call queue limit is reached for auth_id {}".format(queue_id))
+                response['message'] = 'Max call queue reached'
+                return jsonify(**response), 429
```
Comment (manish-plivo) [inline sharq_server/server.py:136]: Put some comment in the code to explain the flow

File: sharq_server/server.py [hunk 4]
```diff
@@ -114,14 +112,40 @@ def _view_enqueue(self, queue_type, queue_id):
...
+
+                return jsonify(**response), 201
+            else:
+                """
+                We are blocking calls queuing if they reached a certain limit.
+                limit = account_cps * 6 Hours = account_cps * (6*60*60)
+                Hence blocking the calls and passing this message to customer
+                """
+                print("Max call queue limit is reached for auth_id {}".format(queue_id))
+                response['message'] = 'Max call queue reached'
```
Comment (manish-plivo) [inline sharq_server/server.py:140]: As sharq is suppose to be generic and not just for calls. Let's keep the message generic. 'Max queue length reached.

File: sharq_server/server.py [hunk 5]
```diff
@@ -114,14 +112,40 @@ def _view_enqueue(self, queue_type, queue_id):
...
+                    response['message'] = e.message
+                    return jsonify(**response), 400
+
+                return jsonify(**response), 201
+            else:
+                """
+                We are blocking calls queuing if they reached a certain limit.
+                limit = account_cps * 6 Hours = account_cps * (6*60*60)
+                Hence blocking the calls and passing this message to customer
+                """
```
Comment (manish-plivo) [inline sharq_server/server.py:138]: let's sync on type of comment required

File: sharq_server/server.py [hunk 6]
```diff
@@ -114,14 +112,46 @@ def _view_enqueue(self, queue_type, queue_id):
...
-            response = self.sq.enqueue(**request_data)
-        except Exception as e:
-            traceback.print_exc()
-            response['message'] = e.message
-            return jsonify(**response), 400
-
-        return jsonify(**response), 201
+        max_queued_length = request_data['payload'].get('max_queued_length', None)
+        """
+        If we are passing max_queued_length in the request from api-voice
```
Comment (manish-plivo) [inline sharq_server/server.py:117]: if max_queued_length is present in request param the queue length will be limited to this value

File: sharq_server/server.py (hunk 5 above)
Comment (manish-plivo) [reply sharq_server/server.py:143]: Please remove above comment

File: sharq_server/server.py [hunk 7]
```diff
@@ -114,14 +112,46 @@ def _view_enqueue(self, queue_type, queue_id):
...
+                    return jsonify(**response), 400
+
+                return jsonify(**response), 201
+            else:
+                """
+                We are blocking calls queuing if they reached a certain limit.
+                limit = account_cps * 6 Hours = account_cps * (6*60*60)
+                Hence blocking the calls and passing this message to customer
+                """
+                print("Max call queue limit is reached for auth_id {}".format(queue_id))
```
Comment (manish-plivo) [inline sharq_server/server.py:144]: let's remove this print statement

**Review by manish-plivo — APPROVED** [review]

File: requirements.txt [hunk 8]
```diff
@@ -10,4 +10,4 @@ itsdangerous==0.24
 msgpack==0.5.6
 ujson==2.0.0
 uWSGI==2.0.21
-SharQ==1.2.0
+git+https://plivo-voice-team:b449c79293d33e2da9b7deb4aa877a295bcccc4f@github.com/plivo/sharq@VT-5650
```
Comment (abhishekGupta-Plivo) [inline requirements.txt:13]: do we need master branch here

**Review by manish-plivo — APPROVED** [review]

**Pull request merged**